portal/v2/src/ClusterDetailListComponents/ClusterOperatorList.tsx

```diff
@@ -0,0 +1,188 @@
+import * as React from 'react';
+import { useState, useEffect } from "react"
+import { Stack, StackItem, IconButton, IIconStyles, SelectionMode } from '@fluentui/react';
+import { Link } from '@fluentui/react/lib/Link';
+import { IColumn } from '@fluentui/react/lib/DetailsList';
+import { ShimmeredDetailsList } from '@fluentui/react/lib/ShimmeredDetailsList';
+import { IClusterOperator } from './ClusterOperatorsWrapper';
+import { ClusterOperatorsComponent } from './ClusterOperator';
+
+export declare interface IClusterOperatorList {
+  name: string;
+  available: string;
+}
+
+interface ClusterOperatorComponentProps {
+  clusterOperators: any
+  clusterName: string
+}
+
+export interface IClusterOperatorListState {
+  clusterOperators: IClusterOperator[]
+  clusterName: string
+}
+
+export class ClusterOperatorListComponent extends React.Component<ClusterOperatorComponentProps, IClusterOperatorListState> {
+  
+  constructor(props: ClusterOperatorComponentProps) {
+      super(props)
+
+      this.state = {
+          clusterOperators: this.props.clusterOperators,
+          clusterName: this.props.clusterName,
+      }
+  }
+  
+  
+  public render() {
+    return (
+        <ClusterOperatorListHelperComponent clusterOperators={this.state.clusterOperators} clusterName={this.state.clusterName}/>
+      )
+  }
+}
+
+export function ClusterOperatorListHelperComponent(props: {
+     clusterOperators: any,
+     clusterName: string
+}) {
+    const [columns, setColumns] = useState<IColumn[]>([
+    {
+      key: "clusterOperatorName",
+      name: "Name",
+      fieldName: "name",
+      minWidth: 150,
+      maxWidth: 350,
+      isResizable: true,
+      isPadded: true,
+      showSortIconWhenUnsorted: true,
+      isSortedDescending: false,
+      isSorted: true,
+      onRender: (item: IClusterOperatorList) => (
+        <Link onClick={() => _onClusterOperatorInfoLinkClick(item.name)}>{item.name}</Link>
+      ),
+    },
+    {
+      key: "clusterOperatorAvailable",
+      name: "Available",
+      fieldName: "available",
+      minWidth: 50,
+      maxWidth: 50,
+      isPadded: true,
+      isResizable: true,
+      isSortedDescending: false,
+      isSorted: true,
+      showSortIconWhenUnsorted: true,
+    }
+  ])
+
+  const [clusterOperatorList, setClusterOperatorList] = useState<IClusterOperatorList[]>([])
+  const [clusterOperatorDetailsVisible, setClusterOperatorDetailsVisible] = useState<boolean>(false)
+  const [currentClusterOperator, setCurrentclusterOperator] = useState<string>("")
+  const [shimmerVisibility, SetShimmerVisibility] = useState<boolean>(true)
+
+  useEffect(() => {
+    setClusterOperatorList(createClusterOperatorList(props.clusterOperators))
```

Comment:
You might not need the `useEffect` here. This code will cause a double-render (this will show when developing with React strict mode on): https://reactjs.org/docs/hooks-faq.html#how-do-i-implement-getderivedstatefromprops

Reply:
Fixed